### PR TITLE
labelFunction now formats tooltips

### LINKED ISSUE #196: formatter for tooltip in axes mode
Whate about formatter function for tooltip in axes mode? It does not work!

### PR DESCRIPTION
**Pull request by chaosmail**
Hi n3-charts team,

Here is a PR the allows the `labelFunction` to format the tooltip values as well; here a link to the demo: [http://plnkr.co/edit/naTVhb6JJrSKm9BL3Rg5](http://plnkr.co/edit/naTVhb6JJrSKm9BL3Rg5?p=preview)

There are 2 more fixes in this PR, that I had to implement to make the tests pass:
* Columns event handlers now bound to *rect* nodes instead of *g* nodes
* Missing options are now overwritten with the default option values

I also have a proposal to rename the `labelFunction` to `ticksFormatter` - a much more descriptive name; like this we could also add `tooltipFormatter` to have finer control over the formatting. I could also think of `ticksFormat` and `tooltipFormat` options to accept d3.format strings.

Fixes #196

Best,
Christoph

### CODE REVIEW

**Comment by lorem--ipsum** [comment]
Great code, can you rebase on master please ?

**Comment by chaosmail** [comment]
What about the proposed change? I would like to mark the `labelFunction` option as deprecated and introduce  the more descriptive `ticksFormatter` and `tooltipFormatter` functions as well as `ticksFormat` and `tooltipFormat` (string for d3.format /d3.time.format). Is this change welome?

**Comment by lorem--ipsum** [comment]
Yeah, sounds good. I like the more explicit names you came with !

**Comment by chaosmail** [comment]
I also added a demo showing the new features: [http://plnkr.co/edit/FmYY5GJT7hgEJ13tIf3y](http://plnkr.co/edit/FmYY5GJT7hgEJ13tIf3y?p=preview)

**Comment by lorem--ipsum** [comment]
Cool ! Meanwhile, I've made a release script that bumps the version and everything. Can you rebase on master please ?

File: README.md [hunk 1]
```diff
@@ -71,7 +71,10 @@ The `axes` keys can be undefined. Otherwise, it can contain an `x̀` key with th
 
  + `key` : optional, defines where the chart will look for abscissas values in the data (default is 'x').
  + `type` : optional, can be either 'date' or 'linear' (default is 'linear'). If set to 'date', the chart will expect Date objects as abscissas. No transformation is done by the chart itself, so the behavior is basically D3.js' time scale's.
- + `labelFunction` : optional, allows to format the axis' ticklabels. Must be a function that accepts a single argument and returns a string.
+ + `ticksFormat` : optional, format string, that is parsed with d3.format (d3.time.format for type 'date') to format the axis' ticklabels
```
Comment (lorem--ipsum) [inline README.md]: Could you add the default behavior as well ? Like "default is raw string" or something ?

**Comment by lorem--ipsum** [comment]
Also, I might be wrong but I think you forgot to update the scrubber code so that it uses the new tooltip format function ?

**Comment by chaosmail** [comment]
The scrubber uses its own formatter function defined in the tooltip options. I think it would be nice, if per default, the scrubber shows the same format "x: y" as defined in the tooltipFormat (for x and y values)

**Comment by lorem--ipsum** [comment]
Yeah, I didn't dive into the code and just looked at your demo while fiddling with a couple of parameters ^^ Well eventually we'll make the scrubber behaves like the other tooltip mode. At first I was even thinking about removing the `axes` mode. Anyway.

Thanks for your contributions, I appreciate. I you wanna shoot me an email sometime, I'd be interested in knowing how you plan to use this little project :-)  vaporzub [@] gmail [.] com

**Comment by lorem--ipsum** [comment]
:poultry_leg:

**Pull request merged**